src/huggingface_hub/hf_file_system.py

```diff
@@ -413,7 +413,21 @@ class HfFileSystemFile(fsspec.spec.AbstractBufferedFile):
     def __init__(self, fs: HfFileSystem, path: str, revision: Optional[str] = None, **kwargs):
         super().__init__(fs, path, **kwargs)
         self.fs: HfFileSystem
-        self.resolved_path = fs.resolve_path(path, revision=revision)
+
+        mode = kwargs.get("mode", "r")
```

Comment:
`fsspec`'s `AbstractBufferedFile` has "rb" as the default mode...

Reply:
Oh ok, thanks for letting me know. I'll open a follow-up PR to update that but in any case it doesn't change anything (this value is used only to check if it's `"w"` or `"wb"`)

Reply:
I've made this commit (https://github.com/huggingface/huggingface_hub/commit/18d0ae2ebfcfc696c50a67e7dad8fa63f5f23068) to avoid confusion